uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/task.js

```diff
@@ -1,271 +1,294 @@
+            'options': {
+              'limit': 0
+            },
+            'is_current_revision': '1',
+            'is_deleted': '0',
+            'sequential': '1',
+            'return': 'activity_date_time, activity_type_id, assignee_contact_id, details, id, source_contact_id, target_contact_id, subject, status_id'
+          }, params);
+
+          Task.get({
+            json: params
+          }, function(data) {
+            deferred.resolve(data.values);
+          }, function() {
+            deferred.reject('Unable to fetch tasks list');
+          });
+
+          return deferred.promise;
+        },
+        getOptions: function() {
+          var deferred = $q.defer(),
+            deferredTaskType = $q.defer(),
+            deferredTaskStatus = $q.defer(),
+            taskType = {
+              arr: [],
+              obj: {}
+            },
+            taskStatus = {
+              arr: [],
+              obj: {}
+            }
+
+          Task.get({
+            action: 'getoptions',
+            json: {
+              'field': 'activity_type_id',
+              'options': {
+                'limit': 0
+              }
+            }
+          }, function(data) {
+            var optionId;
+
+            for (optionId in data.values) {
+              taskType.arr.push({
+                key: optionId,
+                value: data.values[optionId]
+              })
+            }
+
+            taskType.obj = data.values;
+
+            deferredTaskType.resolve(taskType);
+          });
+
+          Task.get({
+            action: 'getoptions',
+            json: {
+              'field': 'status_id'
+            }
+          }, function(data) {
+            var optionId;
+
+            for (optionId in data.values) {
+              taskStatus.arr.push({
+                key: optionId,
+                value: data.values[optionId]
+              })
+            }
+
+            taskStatus.obj = data.values;
+
+            deferredTaskStatus.resolve(taskStatus);
+          });
+
+          $q.all({
+            taskType: deferredTaskType.promise,
+            taskStatus: deferredTaskStatus.promise
+          }).then(function(options) {
+            deferred.resolve(options);
+          });
+
+          return deferred.promise;
+        },
+        save: function(task) {
+
+          if (!task || typeof task !== 'object') {
+            return null;
+          }
+
+          if (task.activity_date_time instanceof Date) {
+            // convert to format readable by backend
+            task.activity_date_time = moment(task.activity_date_time.getTime()).format('YYYY-MM-DD');
+          }
+
+          var deferred = $q.defer(),
+            params = angular.extend({
+              sequential: 1,
+              debug: config.DEBUG
+            }, task),
+            val;
+
+          Task.save({}, {
+            action: 'create',
+            entity: 'Task',
+            sequential: 1,
+            debug: config.DEBUG,
```

Comment:
No it doesn't need to, rather its appropriate to send in request body. I prefer to add only required data as query string that is `action` and `entity`, and send every other data in request body.